### ISSUE TITLE
NPC Missions not being marked as complete

### ISSUE THREAD

**Issue by Berkyjay** [CleverRaven/Cataclysm-DDA#34789]
# Describe the bug
There are two issues that I see:
1. After getting a mission from an NPC I will lose the mission dialog options (i.e. "Have you found it yet?").  I'm not sure what is causing this to happen.  But eventually I notice that the dialog options disappears.

2. While I have a mission from an NPC, the dialog option "Can I help you with anything" is available.  This allows me to get another mission from the NPC, which will prevent me from completing the first mission they gave me.

I'm unsure if these issues are related.  I have tried using the debug menu option to complete the bugged missions but nothing happens when I do.

# Steps To Reproduce

I can't faithfully recreate the issue.  It seems to happen after a while into the game.  It also seems to happen more when I have multiple NPCs following me with multiple missions queued up.

# Expected behavior

Missions get marked as completed.  No new missions can be given before the prior mission is completed.  The ability to debug missions as complete.

# Screenshots

![NPCDebug_01](https://user-images.githubusercontent.com/5049386/66864256-9308f680-ef49-11e9-8d4e-b49dcd11bd65.PNG)

![NPCDebug_02](https://user-images.githubusercontent.com/5049386/66864260-94d2ba00-ef49-11e9-8359-4f2d2087c2cc.PNG)

![NPCDebug_03](https://user-images.githubusercontent.com/5049386/66864264-969c7d80-ef49-11e9-89cc-eca2f2c994dd.PNG)

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1809
- Game Version: 0.D-8520-gcb7938e [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Magiclysm [magiclysm],
    Bionic Slots [cbm_slots],
    Stats Through Skills [StatsThroughSkills]
]

**Comment by benjaminmkos** [comment]
Chiming in to say that I've seen the same in my current save. Three missions from the same NPC, no "check in" dialogue for any of them.

- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.D-8521-g0c7d821 [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Rail Stations [No_Rail_Stations]
]

I have also found [this old issue](https://github.com/CleverRaven/Cataclysm-DDA/issues/23824) which seems to indicate that NPC skill training breaks mission progression.

**Comment by Berkyjay** [comment]
Yup, I can confirm that asking for skill training breaks the missions for NPCs.

I spawned 2 NPCs, got missions from them both.  Spawned the quest items, completed the missions as normal.  But when I repeat this process but ask for training from the NPCs before completing the missions, I lose the mission dialog.  Plus, I am able to ask for the next mission without completing the first mission.  

This seems like a duplicate bug and I'll post over on the bug @benjaminmkos provided.

**Comment by jkraybill** [comment]
#38632 does seem to fully fix the issue where skills training resets a mission, but I'm not sure that fully resolves this issue.

If anybody can come up with a reproduction scenario where missions get forgotten without skill training happening, please log that! Also I still don't have a theory on what causes #38325 which is a similar symptom.

**Comment by KorGgenT** [comment]
either repro scenario or a savefile with the broken behavior would be amazing!

**Comment by benjaminmkos** [comment]
So its been quite some time since I paid any attention to this but this should be my save from when I found this behavior: [NPC Missions Bug.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4367047/NPC.Forgets.Missions.zip)

In the Missions screen there are three active missions from the same NPC, and if you try to talk to her (she's in the room with you when you load the save) she won't have "check in" dialogue for any of them. 

Due to this nature of this bug I was never able to home in on concrete reproduction steps other than the skill training so @jkraybill 's PR referenced above may be a complete fix but it's hard to say.
